languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -8788,6 +8788,30 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id= "CA_VA_MAL_SE_FINIR" name="ça va mal se finir" type="style" default="temp_off">
+          <rule>
+              <pattern>
+                  <token regexp="yes">ça</token>
+                  <token regexp="yes">va</token>
+                  <token regexp="yes">mal</token>
+                  <marker><token regexp="yes">se</token></marker>
+                  <token regexp="yes">finir</token>
+              </pattern>
+              <message>Dans cette expression française familière,« se finir » est une erreur grammaticale. Nous vous suggérons  d’écrire « ça va mal <suggestion><match no="4" regexp_match="se" regexp_replace=""/></suggestion> finir ». </message>
+              <example correction="">Je pense que ça va mal <marker>se</marker> finir.</example>
```

Comment:
It would be nice to help justify the rule with a <url>…</url> if there is a relevant authoritative link.